mmeval/metrics/coco_panoptic.py

```diff
@@ -0,0 +1,660 @@
+        self._results = cache_results
+        self._coco_api = cache_coco_api
+
+        return metric_result
+
+    def default_dump(self, obj):
+        """Convert numpy classes to JSON serializable objects."""
+        if isinstance(obj, (np.integer, np.floating, np.bool_)):
+            return obj.item()
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        else:
+            return obj
+    
+    def parse_pq_results(self, pq_results: dict) -> dict:
+        """Parse the Panoptic Quality results.
+
+        Args:
+            pq_results (dict): Panoptic Quality results.
+
+        Returns:
+            dict: Panoptic Quality results parsed.
+        """
+        result = dict()
+        result['PQ'] = 100 * pq_results['All']['pq']
+        result['SQ'] = 100 * pq_results['All']['sq']
+        result['RQ'] = 100 * pq_results['All']['rq']
+        result['PQ_th'] = 100 * pq_results['Things']['pq']
+        result['SQ_th'] = 100 * pq_results['Things']['sq']
+        result['RQ_th'] = 100 * pq_results['Things']['rq']
+        result['PQ_st'] = 100 * pq_results['Stuff']['pq']
+        result['SQ_st'] = 100 * pq_results['Stuff']['sq']
+        result['RQ_st'] = 100 * pq_results['Stuff']['rq']
+        return result
+
+    def print_panoptic_table(self, pq_results, classwise_results=None):
+        """Print the panoptic evaluation results table.
+
+        Args:
+            pq_results(dict): The Panoptic Quality results.
+            classwise_results(dict | None): The classwise Panoptic Quality results.
+                The keys are class names and the values are metrics.
+        """
+        headers = ['', 'PQ', 'SQ', 'RQ', 'categories']
+        data = [headers]
+        for name in ['All', 'Things', 'Stuff']:
+            numbers = [
+                f'{(pq_results[name][k] * 100):0.3f}' for k in ['pq', 'sq', 'rq']
+            ]
+            row = [name] + numbers + [pq_results[name]['n']]
+            data.append(row)
+        table = AsciiTable(data)
+        print('Panoptic Evaluation Results:\n' + table.table)
+
+        if classwise_results is not None:
+            class_metrics = [(name, ) + tuple(f'{(metrics[k] * 100):0.3f}'
+                                            for k in ['pq', 'sq', 'rq'])
+                            for name, metrics in classwise_results.items()]
+            num_columns = min(8, len(class_metrics) * 4)
+            results_flatten = list(itertools.chain(*class_metrics))
+            headers = ['category', 'PQ', 'SQ', 'RQ'] * (num_columns // 4)
+            results_2d = itertools.zip_longest(
+                *[results_flatten[i::num_columns] for i in range(num_columns)])
+            data = [headers]
+            data += [result for result in results_2d]
+            table = AsciiTable(data)
+            print(msg='Classwise Panoptic Evaluation Results:\n' + table.table)
+
+#TODO: 写一下这里的return结果
+    def pq_compute_single_core(self, proc_id,
```

Comment:
The `pq_compute_single_core` and `pq_compute_multi_core` are modified from `https://github.com/cocodataset/panopticapi/blob/master/panopticapi/evaluation.py`, it is better to move them into `mmeval/metrics/_vendor`~